src/io/bin.cpp

```diff
@@ -593,15 +593,15 @@ namespace LightGBM {
   }
 
   void BinMapper::SaveBinaryToFile(const VirtualFileWriter* writer) const {
-    writer->AlignedWrite(&num_bin_, sizeof(num_bin_));
-    writer->AlignedWrite(&missing_type_, sizeof(missing_type_));
-    writer->AlignedWrite(&is_trivial_, sizeof(is_trivial_));
+    writer->Write(&num_bin_, sizeof(num_bin_));
+    writer->Write(&missing_type_, sizeof(missing_type_));
+    writer->Write(&is_trivial_, sizeof(is_trivial_));
     writer->Write(&sparse_rate_, sizeof(sparse_rate_));
-    writer->AlignedWrite(&bin_type_, sizeof(bin_type_));
+    writer->Write(&bin_type_, sizeof(bin_type_));
     writer->Write(&min_val_, sizeof(min_val_));
     writer->Write(&max_val_, sizeof(max_val_));
-    writer->AlignedWrite(&default_bin_, sizeof(default_bin_));
-    writer->AlignedWrite(&most_freq_bin_, sizeof(most_freq_bin_));
+    writer->Write(&default_bin_, sizeof(default_bin_));
+    writer->Write(&most_freq_bin_, sizeof(most_freq_bin_));
```

Comment:
I think this worked!!! I was able to get the UBSAN tests to fail with this change

![image](https://user-images.githubusercontent.com/7608904/95409291-e9074480-08e6-11eb-8e14-d5426a6fadb9.png)

https://github.com/microsoft/LightGBM/pull/3439/checks?check_run_id=1223773068